src/custom_request_schedule_manager.cc

```diff
@@ -118,14 +100,35 @@ CustomRequestScheduleManager::GetScheduleFromDataset() const
 
   const size_t dataset_size{data_loader_->GetTotalSteps(0)};
 
-  for (size_t dataset_index{0}; dataset_index < dataset_size; ++dataset_index) {
+  if (dataset_size == 0) {
+    throw std::runtime_error(
+        "Expected input data JSON to have at least one request payload.");
+  }
+
+  if (warmup_request_count_ >= dataset_size) {
+    throw std::runtime_error(
+        "Expected warmup request count to be less than the dataset size.");
+  }
+
+  if (dataset_size != warmup_request_count_ + benchmark_request_count_) {
+    throw std::runtime_error(
+        "Expected input data JSON size to be equal to the warmup request count "
+        "plus the benchmark request count.");
+  }
+
+  for (size_t dataset_index{0}; dataset_index < warmup_request_count_;
+       ++dataset_index) {
     const auto timestamp{GetTimestamp(dataset_index)};
-    schedule.push_back(timestamp);
+    warmup_schedule.push_back(timestamp);
   }
 
-  std::sort(schedule.begin(), schedule.end());
+  for (size_t dataset_index{warmup_request_count_};
+       dataset_index < dataset_size; ++dataset_index) {
+    const auto timestamp{GetTimestamp(dataset_index)};
+    benchmark_schedule.push_back(timestamp);
+  }
```

Comment:
This is the main magic of the implementation.

Two schedules get created instead of one.

First one is the warmup schedule up to the `warmup_request_count_`-th payload.

Second one is benchmark schedule starting at `warmup_request_count_+1`-th payload up to the last.